File: backup/src/jclang/Cursor.java


package jclang;

import org.jetbrains.annotations.NotNull;
import jclang.structs.CXCursor;
import jclang.structs.CXString;

public class Cursor {
    private final CXCursor.ByValue cursor;

    public Cursor(@NotNull CXCursor.ByValue cursor) {
        this.cursor = cursor;
    }

    @NotNull
    public String getSpelling() {
        CXString.ByValue spelling = LibClang.I.getCursorSpelling(cursor);
        NativePool.I.record(spelling);
        return LibClang.I.getCString(spelling);
    }

    @NotNull
    public CursorKind getKind() {
        // We should've called clang_getCursorKind here, but this works and is more efficient
        return CursorKind.fromNative(cursor.kind);
    }

    @NotNull
    public Type getType() {
        return new Type(LibClang.I.getCursorType(cursor));
    }
}

File: libclang/src/main/java/klang/jvm/Cursor.kt
package klang.jvm

import klang.jvm.binding.CXCursor

class Cursor(private val cursor: CXCursor.CXCursorByValue) {
    val spelling: String
        get() = Clang.getCursorSpelling(cursor)
    val kind: CursorKind // We should've called clang_getCursorKind here, but this works and is more efficient
        get() = CursorKind.fromNative(cursor.kind)
    val type: Type
        get() = Type(Clang.getCursorType(cursor))
}